backend/src/channel/channel.service.ts

```diff
@@ -0,0 +1,512 @@
+            where: {
+                id: channel_id,
+            },
+            include: {
+                ChannelMembers: true,
+            },
+        });
+        if (!channel) {
+            return 'Channel not found';
+        }
+        const user = await this.searchUserById(req.user['id']);
+        if (!user) {
+            return 'User not found';
+        }
+        const channelMember = await prisma.channelMembers.findFirst({
+            where: {
+                channel_id: channel_id,
+                user_id: user.id,
+            },
+        });
+        if (!channelMember) {
+            return 'You are not a member of the channel';
+        }
+        if (channelMember.MemberType !== 'OWNER' && channelMember.MemberType !== 'ADMIN') {
+            return 'You are not the owner of the channel';
+        }
+        const channelMember2 = await prisma.channelMembers.findFirst({
+            where: {
+                channel_id: channel_id,
+                user_id: user_id,
+            },
+        });
+        if (!channelMember2 && action === 'invite') {
+            await this.createChannelMember(user_id, channel_id, 'INVITED');
+            return 'User invited';
+        }
+
+        else if (channelMember2) {
+            switch (action) {
+                case 'ban':
+                    if (channelMember2.MemberType === 'BANNED') {
+                        return 'User is already banned';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'BANNED');
+                    return 'User banned';
+                case 'mute':
+                    if (channelMember2.MemberType === 'MUTED') {
+                        return 'User is already muted';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'MUTED');
+                    return 'User muted';
+                case 'kick':
+                    if (channelMember2.MemberType === 'NOTMEMBER') {
+                        return 'User is not a member of the channel';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'NOTMEMBER');
+                    return 'User kicked';
+                case 'unban':
+                    if (channelMember2.MemberType !== 'BANNED') {
+                        return 'User is not banned';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'MEMBER');
+                    return 'User unbanned';
+                case 'unmute':
+                    if (channelMember2.MemberType !== 'MUTED') {
+                        return 'User is not muted';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'MEMBER');
+                    return 'User unmuted';
+                case 'accept':
+                    if (channelMember2.MemberType !== 'REQUESTED') {
+                        return 'User is not requested';

```

Comment:
the return statement is misleading use this instead: return 'User is not requesting to join'